sample/buildSrc/src/main/kotlin/com/dropbox/sample/tasks/AffectedTasksPlugin.kt

```diff
@@ -0,0 +1,130 @@
+package com.dropbox.affectedmoduledetector.tasks
+
+import com.dropbox.affectedmoduledetector.AffectedModuleDetector
+import com.dropbox.affectedmoduledetector.DependencyTracker
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.testing.Test
+import java.util.*
+
+var TEST_TASK_TO_RUN_EXTENSION = "TestTasks"
+
+open class TestTasks {
+    var variantToTest = "debug"
+    val assembleAndroidTestTask = "assemble${variantToTest.capitalize()}AndroidTest"
+    val runAndroidTestTask = "connected${variantToTest.capitalize()}AndroidTest"
+    val jvmTest = "test${variantToTest.capitalize()}UnitTest"
+    val jvmTestBackup = "test"
+}
+
+/**
+ * unlike [AffectedTestsPlugin] which skips unaffected tests, this plugn instead creates a task and
+ * registers all affected test tasks. Advantage is speed in not needing to skip modules at a large scale
+ *
+ * Registers 3 tasks
+ * gradlew runAffectedUnitTests - runs jvm tests
+ * gradlew runAffectedAndroidTests - runs connected tests
+ * gradlew assembleAffectedAndroidTests - assembles but does not run on device tests, useful when working with device labs
+ */
+class AffectedTasksPlugin : Plugin<Project> {
+    var ANDROID_TEST_BUILD_VARIANT = "AndroidTest"
+
+    lateinit var testTasks: TestTasks
+    override fun apply(project: Project) {
+        project.extensions.add(TEST_TASK_TO_RUN_EXTENSION, TestTasks())
+        project.afterEvaluate {
+            val rootProject = project.rootProject
+            testTasks = requireNotNull(
+                project.extensions.findByName(TEST_TASK_TO_RUN_EXTENSION)
+            ) as TestTasks
+            registerAffectedTestTask(
+                "runAffectedUnitTests",
+                testTasks.jvmTest, testTasks.jvmTestBackup, rootProject,
+            )
+
+            registerAffectedAndroidTests(rootProject)
+            registerAffectedConnectedTestTask(rootProject)
+        }
+
+        filterAndroidTests(project)
```

Comment:
I am not sure if these two methods are necessary for other folks or if they are just unique issues in our environment.   We may want to remove them to avoid confusion.

Reply:
I think it makes sense since it is keyed off them enabling the detector.